packages/headless/src/controllers/commerce/sort/core/headless-core-commerce-sort.ts

```diff
@@ -0,0 +1,168 @@
+import {Schema} from '@coveo/bueno';
+import {AsyncThunkAction} from '@reduxjs/toolkit';
+import {CommerceEngine} from '../../../../app/commerce-engine/commerce-engine';
+import {
+  buildFieldsSortCriterion,
+  buildRelevanceSortCriterion,
+  SortBy,
+  SortByFields,
+  SortByFieldsFields,
+  SortByRelevance,
+  SortCriterion,
+  SortDirection,
+  sortCriterionDefinition,
+} from '../../../../features/commerce/sort/sort';
+import {applySort} from '../../../../features/commerce/sort/sort-actions';
+import {sortReducer as commerceSort} from '../../../../features/commerce/sort/sort-slice';
+import {updatePage} from '../../../../features/pagination/pagination-actions';
+import {loadReducerError} from '../../../../utils/errors';
+import {validateInitialState} from '../../../../utils/validate-payload';
+import {
+  buildController,
+  Controller,
+} from '../../../controller/headless-controller';
+
+export type {SortByRelevance, SortByFields, SortByFieldsFields, SortCriterion};
+export {
+  SortBy,
+  SortDirection,
+  buildRelevanceSortCriterion,
+  buildFieldsSortCriterion,
+};
+
+export interface SortProps {
```

Comment:
Declaring this interface here allows the PLP and Search sort controllers to reuse it.